version: '3.8'
services:

  web:
    build: .
    command: >
      sh -c "python manage.py migrate &&
             python manage.py runserver 0.0.0.0:8080"
    volumes:
      - .:/code
    ports:
      - "8080:8080"
    env_file:
      - .env
    stdin_open: true
    tty: true
    depends_on:
      - db
    networks:
      - backend

  db:
    container_name: celery_postgres_db_local
    image: postgres:14.1-alpine
    env_file:
      - .env
    ports:
      - '5434:5432'
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
    networks:
      - backend

  celery:
    build: .
    command: celery -A django_celery_app worker --loglevel=info
    env_file:
      - .env
    volumes:
      - .:/code
    depends_on:
      - web
      - db
    networks:
      - backend

  # celery-beat:
  #   build: .
  #   command: celery -A django_celery_app beat --loglevel=info
  #   volumes:
  #     - .:/code
  #   depends_on:
  #     - web
  #     - db
  #   networks:
  #     - backend

  redis:
    image: redis:6
    networks:
      - backend

networks:
  backend:
volumes:
  postgres_data:
